### DISCUSSION TITLE
(Livewire v2) Livewire not found

### DISCUSSION THREAD

**Discussion by nixaura** [livewire/livewire#6900] [1. Help]
Hi there. I'm having an issue with the @livewireScripts not loading in time(?) and I need some help.

In my main blade template I have:

```
<!DOCTYPE html>
    <html lang="en">
        <head>
            <!-- Title -->
            <title>lllllllllllll</title>

            <!-- @livewireStyles is the last line of partials.masterStyle -->
            @include('partials.masterStyle')

            @include('theme.default')
            
        </head>

        <body>
              ....
            @include('partials.topNavBar')
            @livewire('admin.view-map', ['showCoordinateMarker' => false, 'applet' => false])
              ....
            @livewire('public.reserve-modal')
            @livewireScripts
        </body>
</html>
                    
```

The @livewireScripts is the last line before the <body> close tag. 

Within the @livewire('admin.view-map') component I have a <script> tag with:
```
<script>
        function checkTime() {
            console.log("Calling updateTimeOfDay");
            Livewire.emit('updateTimeOfDay');
            smoothlyShiftColorAndOpacity();
            .....
         }
</script>
```
            
When I load the page, I am receiving the error message below that is referenced in the JavaScript above.

`Uncaught ReferenceError: Livewire is not defined at checkTime`

Could I get some direction on where I could troubleshoot?

Thanks!

**Comment by lrljoe** [comment]
Livewire isn't ready at the time that your checkTime() function is being called, so your browser doesn't know what Livewire.X is.

The docs available here do explain this:
[https://laravel-livewire.com/docs/2.x/inline-scripts](https://laravel-livewire.com/docs/2.x/inline-scripts)

```
<div>
    <!-- Your components HTML -->
 
    <script>
        document.addEventListener('livewire:load', function () {
            // Your JS here.
        })
    </script>
</div>
```
